workspaces/demo-app/src/components/cogito-address/CogitoAddress.test.js

```diff
@@ -145,58 +179,84 @@ describe('CogitoAddress', () => {
 
     it('sets user identity and connection status in the redux store', async () => {
       const { getByText, store } = render(cogitoAddress())
-      const readButton = await waitForElement(() => getByText(/read your identity.../i))
+      const readButton = await waitForElement(() =>
+        getByText(/read your identity.../i)
+      )
       fireEvent.click(readButton)
-      const doneButton = getByText(/done/i)
-      fireEvent.click(doneButton)
-      await wait(() => expect(store.getState().userData).toMatchObject(defaultIdentity))
+      fakeConnectionSetupDone()
+      await wait(() =>
+        expect(store.getState().userData).toMatchObject(defaultIdentity)
+      )
     })
 
-    it('creates a new telepath channel if user explicitely requests a new QR Code', async () => {
-      const { id: initialId, key: initialKey } = telepathChannel
-      const { getByText } = render(cogitoAddress())
-      const showQRCodeButton = await waitForElement(() => getByText(/show qr code/i))
-      fireEvent.click(showQRCodeButton)
+    describe('when explicitly requesting a new QR code', () => {
+      let getByText
+      let getByTestId
+      let store
+      let rerenderFromReactTestingLibrary
+
+      const showQRCode = async () => {
+        const showQRCodeButton = await waitForElement(() =>
+          getByText(/show qr code/i)
+        )
+        fireEvent.click(showQRCodeButton)
+        await waitForElement(() => getByText(/scan the qr code/i))
+      }
 
-      await wait(() => {
-        expect(telepathChannel.id).not.toEqual(initialId)
-        expect(telepathChannel.key).not.toEqual(initialKey)
+      const readIdentity = async () => {
+        const readButton = await waitForElement(() =>
+          getByText(/read your identity.../i)
+        )
+        fireEvent.click(readButton)
+        fakeConnectionSetupDone()
+      }
+
+      beforeEach(async () => {
+        ;({
```

Comment:
Wondering about the semicolon? See [prettier](https://prettier.io/docs/en/rationale.html#semicolons) and [standardjs](https://standardjs.com/rules.html#semicolons).